### ISSUE TITLE
Accuracy issue in conversion of T5 model.

### ISSUE THREAD

**Issue by VikasOjha666** [NVIDIA/TensorRT#1794]
@nvpohanh 
@rajeevsrao 
@vinhngx 
@joohoonl-nv 
I converted the valhalla/t5-small-qa-qg-hl model from https://github.com/patil-suraj/question_generation repo using the demo/HuggingFace code. I made the necessary changes in the T5ModelConfig file. I verified the predictions from the ONNX model generated in the step. The prediction was good for the ONNX model. But when I converted the model to TensorRT and made a prediction. The prediction is coming as it is random. The conversion is working for the translation model. Please help.

**Comment by vinhngx** [comment]
Does this model use the same vocabulary as the original HF model? If not you may need to fix the vocab size.

**Comment by VikasOjha666** [comment]
Thanks very much for reply @vinhngx . I have changed the vocab size in T5ModelConfig file according to the tokenizer I am using in this model. The ONNX model was giving correct prediction.

**Comment by vinhngx** [comment]
Can you check if this is related? https://github.com/NVIDIA/TensorRT/issues/1613
In particular: 

```
As it turns out, the vocabulary size of t5 is [hardcoded in the trt.py](https://github.com/NVIDIA/TensorRT/blob/9ec6eb6db39188c9f3d25f49c8ee3a9721636b56/demo/HuggingFace/T5/trt.py#L154).
When this is set to the correct vocabulary size of the model, the outputs are as expected.
```

**Comment by VikasOjha666** [comment]
Yes the problem is similar. Please let me check the issue and comment if it works.

**Comment by VikasOjha666** [comment]
Thanks a ton, @vinhngx. The problem was because of the wrong vocab size I was using. I was using the vocab size I got by the tokenizer.vocab_size function which was coming to be 32100 but when I checked the model config I got the size as 32102. After replacing it in the T5ModelConfig file the issue was resolved as in the latest commit trt.py didn't have hardcoded vocab. I am closing this issue as it is resolved. Once again thanks and well wishes for you and your family.
